### PR TITLE
Testing Merging circles polyunion

### PR DESCRIPTION
**Pull request by AbraaoAlves**
With these tests we can see polyunion failing in some cases, for example: 

- for circles polygons with 17 or more steps or lower than 16

### CODE REVIEW

**Comment by juanpujol** [comment]
Thanks @AbraaoAlves will see if I can test and fix polygons before merging.

Here's a quick idea:

```ts
interface GeoJSONFeature {
    type: string;
    geometry: {
        type: string;
        coordinates: number[][][]; // For Polygon
    };
}

// Example GeoJSON data
const exampleGeoJSON: GeoJSONFeature = {
    type: "Feature",
    geometry: {
        type: "Polygon",
        coordinates: [
            [
                [0, 0],
                [0, 1],
                [1, 1],
                [1, 0],
                [0, 0] // Closing point should be the same as the starting point
            ]
        ]
    }
};

function isPolygonClosed(coordinates: number[][][]): boolean {
    const firstPoint = coordinates[0][0];
    const lastPoint = coordinates[0][coordinates[0].length - 1];
    // Check if they are the same
    return firstPoint[0] === lastPoint[0] && firstPoint[1] === lastPoint[1];
}

function fixClosedPolygon(feature: GeoJSONFeature): void {
    if (!isPolygonClosed(feature.geometry.coordinates)) {
        const firstPoint = feature.geometry.coordinates[0][0];
        feature.geometry.coordinates[0].push(firstPoint); // Append first point to close the polygon
        console.log("Polygon was not closed. Fixed.");
    } else {
        console.log("Polygon is already closed.");
    }
}

// Testing
console.log("Before fixing:");
console.log(JSON.stringify(exampleGeoJSON, null, 2));
fixClosedPolygon(exampleGeoJSON);
console.log("After fixing:");
console.log(JSON.stringify(exampleGeoJSON, null, 2));
```

**Comment by juanpujol** [comment]
Ok, so I found the problem. It turns out the intersect check was throwing an error. But, if this fails doesn't mean it won't work, we should skip the merge for that instance. 

I've added a try catch block to avoid throwing and console.error instead.

Also, change the directory structure and simplified the tests.

@AbraaoAlves what do you think?

File: src/index.ts [hunk 1]
```diff
@@ -82,16 +82,22 @@ function polyunionPass(
...
-				mergeCandidate = union(mergeCandidate, targetFeature) as Feature<Polygon>;
-				processed.add(candidate.id);
+			// Attempt to merge the two features if they intersect or overlap
+			try {
+				if (
+					intersect(mergeCandidate, targetFeature) ||
+					booleanOverlap(mergeCandidate, targetFeature)
+				) {
+					mergeCandidate = union(mergeCandidate, targetFeature) as Feature<Polygon>;
+					processed.add(candidate.id);
```
Comment (lamartinecabral) [inline src/index.ts:92]:
```suggestion
					const polygon = union(mergeCandidate, targetFeature) as Feature<Polygon>;
					if (polygon.geometry.type === 'Polygon') {
						mergeCandidate = polygon;
						processed.add(candidate.id);
					}
```

Comment (lamartinecabral) [reply src/index.ts:92]: I found out that when two polygons touch in a vertex, turf does not merge them. Instead, it returns a MultiPolygon.

**Comment by lamartinecabral** [comment]
> I've added a try catch block to avoid throwing and console.error instead.

@juanpujol I found out that another way to avoid throwing is to reduce the precision of the coordinates. Something like `+value.toFixed(13)` should be enough

**Comment by juanpujol** [comment]
@lamartinecabral, if you have an improvement feel free to create an issue with a new PR. Right now, this PR fixes the main issue we had to deploy the lib to production. This works for me and if it works for you, please approve it and we can make it better on the next patch.

Thank you.

**Review by juanpujol — APPROVED** [review]

**Pull request merged**